qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -1,18 +1,332 @@
+                           parameter_class=GroupParameter,
+                           vals=vals.Numbers(0.001, 80),
+                           docstring=textwrap.dedent("""
+        Power compliance (in W). Resolution: 0.001 W. If it is not entered, 
+        the power compliance is not set. This parameter is not available for
+        HVSMU. 0.001 to 2 for MPSMU/HRSMU, 0.001 to 20 for HPSMU, 0.001 to 
+        40 for HCSMU, 0.001 to 80 for dual HCSMU, 0.001 to 3 for MCSMU, 
+        0.001 to 100 for UHVU
+                           """))
+
+        self.add_parameter(name='chan',
+                           initial_value=self.parent.channels[0],
+                           parameter_class=GroupParameter)
+
+        self._set_sweep_steps_group = Group(
+            [self.chan,
+             self.sweep_mode,
+             self.sweep_range,
+             self.sweep_start,
+             self.sweep_end,
+             self.sweep_steps,
+             self.current_compliance,
+             self.power_compliance],
+            set_cmd='WV '
+                    '{chan},'
+                    '{sweep_mode},'
+                    '{sweep_range},'
+                    '{sweep_start},'
+                    '{sweep_end},'
+                    '{sweep_steps},'
+                    '{current_compliance},'
+                    '{power_compliance}',
+            get_cmd=self._get_sweep_steps(),
+            get_parser=self._get_sweep_steps_parser)
+
+
+    @staticmethod
+    def _get_sweep_delays() -> str:
+        msg = MessageBuilder().lrn_query(
+            type_id=constants.LRN.Type.STAIRCASE_SWEEP_MEASUREMENT_SETTINGS
+        )
+        cmd = msg.message
+        return cmd
+
+    @staticmethod
+    def _get_sweep_delays_parser(response: str) -> Dict[str, float]:
+        match = re.search('WT(?P<hold_time>.+?),(?P<delay>.+?),'
+                          '(?P<step_delay>.+?),(?P<trigger_delay>.+?),'
+                          '(?P<measure_delay>.+?)(;|$)',
+                          response)
+        if not match:
+            raise ValueError('Sweep delays (WT) not found.')
+
+        resp_dict = match.groupdict()
+        out_dict = {key: float(value) for key, value in resp_dict.items()}
+        return out_dict
+
+    def _set_sweep_auto_abort(self, val: Union[bool, constants.Abort]):
+        msg = MessageBuilder().wm(abort=val)
+        self.write(msg.message)
+
+    def _set_post_sweep_voltage_condition(
+            self, val: Union[constants.WM.Post, int]):
+        msg = MessageBuilder().wm(abort=self.sweep_auto_abort(), post=val)
+        self.write(msg.message)
+
+    @staticmethod
+    def _get_sweep_steps():
+        msg = MessageBuilder().lrn_query(
+            type_id=constants.LRN.Type.STAIRCASE_SWEEP_MEASUREMENT_SETTINGS
+        )
+        cmd = msg.message
+        return cmd
+
+    @staticmethod
+    def _get_sweep_steps_parser(response: str) -> Dict[
+        str, Union[int, float]]:
+        match = re.search(r'WV(?P<chan>.+?),'
```

Comment:
see whether _get_sweep_steps query returns info from ALL the channels on the B1500. and if so, here we need to take the info only about the current channel.

Reply:
So we can only set WV to one channel hence get query can only return from 1 channel. Howvever this issue got fixed. Most likely with the corrections Jens did.